internal/emscripten/emscripten.go

```diff
@@ -89,8 +109,17 @@ func (v *InvokeFunc) Call(ctx context.Context, mod api.Module, stack []uint64) {
 		panic(err)
 	}
 
-	err = f.CallWithStack(ctx, stack)
+	stackSave, err := mod.ExportedFunction("stackSave").Call(ctx)
 	if err != nil {
 		panic(err)
 	}
+
+	err = f.CallWithStack(ctx, stack)
+	if err != nil {
+		_, _ = mod.ExportedFunction("stackRestore").Call(ctx, stackSave[0])
```

Comment:
I'm not sure we can call this for all errors.
I know that's what the JS code does, but I'm not sure the module is always in a state where we can call this.
Like what happens if your sample calls `exit(1)`, do we go through this? Should we?

Reply:
We could check for specific errors, but I wanted the implementation to be the same as the JS one, and they always call this method, so I assume that the C implementation takes care of things like that.

However, I think calling just `exit(1)` would not make you end up in the error branch of `CallWithStack`

Reply:
Are you sure? Doesn't it produce a `sys.ExitError` here:

	if exitErr, ok := recovered.(*sys.ExitError); ok { // Don't wrap an exit error!
		return exitErr
	}

Reply:
Thanks for checking!